appstudio-controller/controllers/appstudio.redhat.com/promotionrun_controller.go

```diff
@@ -412,10 +420,48 @@ func locateTargetManualBinding(ctx context.Context, promotionRun appstudioshared
 		}
 	}
 
-	return appstudioshared.SnapshotEnvironmentBinding{},
-		fmt.Errorf("unable to locate binding with application '%s' and target environment '%s'",
-			promotionRun.Spec.Application, promotionRun.Spec.ManualPromotion.TargetEnvironment)
+	// Binding not found, so create it
+	components := []appstudioshared.BindingComponent{}
+	componentList := appstudioshared.ComponentList{}
+	if err := k8sClient.List(ctx, &componentList); err != nil {
+		return appstudioshared.SnapshotEnvironmentBinding{}, fmt.Errorf("unable to list components: %v", err)
+	}
+	for _, component := range componentList.Items {
+		if component.Spec.Application == promotionRun.Spec.Application {
+			components = append(components, appstudioshared.BindingComponent{
+				Name: component.Spec.ComponentName,
+			})
+		}
+	}
+	if len(components) == 0 {
+		return appstudioshared.SnapshotEnvironmentBinding{}, fmt.Errorf("unable to find components for application: %s", promotionRun.Spec.Application)
+	}
+	binding := appstudioshared.SnapshotEnvironmentBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      createBindingName(&promotionRun),
+			Namespace: promotionRun.Namespace,
+			Labels: map[string]string{
+				"appstudio.application": promotionRun.Spec.Application,
+				"appstudio.environment": promotionRun.Spec.ManualPromotion.TargetEnvironment,
+			},
+		},
+		Spec: appstudioshared.SnapshotEnvironmentBindingSpec{
+			Application: promotionRun.Spec.Application,
+			Environment: promotionRun.Spec.ManualPromotion.TargetEnvironment,
+			Snapshot:    promotionRun.Spec.Snapshot,
+			Components:  components,
+		},
+	}
+	err := k8sClient.Create(ctx, &binding)
```

Comment:
There are a couple items we can log here:
-  We usually call `LogAPIResourceChangeEvent` whenever we are creating/modifying/deleting a resource, we can dd a call here for that. 
    - These log items are mostly for auditing purposes (allow us to keep a record of changes made to the cluster for security purposes.
- It couldn't hurt to add an INFO-level log `log.Info('Created SnapshotEnvironmentBinding", "application", "(app id), "environment", "environment id")`
    - These log statements are more dev/SRE focus, helping us debug the code.

Reply:
Added logging for these items